### DISCUSSION TITLE
v2021.08.2_23.1_x86-64_0.34

### DISCUSSION THREAD

**Discussion by PartialVolume** [PartialVolume/shredos.x86_64#134] [Announcements]
v2021.08.2_23.1_x86-64_0.34

| Buildroot Version | Kernel Version | Shredos Minor version | Architecture | Nwipe Version |
| ----------------------- | ------------------- | ------------------------------- | ---------------- | ------------------- |
| 2021.08.2             | 5.13.19           |  023                               | x86-64          | 0.34                |

Other software versions used within ShredOS for disc related operations
| hexedit | hdparm | Smartctl |
| ---------- | ---------- | -------------------- |
|    1.5     |   9.62    |           7.2          |

## Notable changes in this release - ShredOS
- Added missing AMDGPU, Radeon and GPU firmware. This was causing some systems to fail to boot to the nwipe screen.

- Increased the vfat disk size from 80Mbytes to ~114MB to accommodate these new additions. This leaves about 11Mbyte free in the VFAT partition for logs and PDF reports. (PDF Reports will be in the v2021.08.2_24_x86-64_0.34 (v24) version.

## Notable changes in this release - Nwipe
No changes. There will be lots of nwipe changes in the v0.35 release. HPA/DCO detection, PDF reports, ATA secure erase.

## Note
- The .img file for writing to a USB flash drive is both legacy (bios) and UEFI bootable.

As this is only a preview for testing booting on previously non bootable devices I'm not planning on creating .iso's or 32bit versions, unless somebody specifically has a system that for some reason they cannot boot using USB.

## Download Counters
| Architecture/format | Download link |
| ------------------------- | --------------------|
|  x86-64 .img for USB | [![GitHub release (by file)](https://img.shields.io/github/downloads/PartialVolume/shredos.x86_64/v2021.08.2_23.1_x86-64_0.34/shredos-2021.08.2_23.1_x86-64_0.34_20230320.img)](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2021.08.2_23.1_x86-64_0.34/shredos-2021.08.2_23.1_x86-64_0.34_20230320.img) |

### 64bit .img for modern Intel & compatible processors post Pentium 4.
```
sha1  188f7e0f0253ca2a8f2d9bce431369abd7413c73  shredos-2021.08.2_23.1_x86-64_0.34_20230320.img
```

<hr /><em>This discussion was created from the release <a href='https://github.com/PartialVolume/shredos.x86_64/releases/tag/v2021.08.2_23.1_x86-64_0.34'>v2021.08.2_23.1_x86-64_0.34</a>.</em>

**Comment by Hellgore1974** [comment]
It doesn't work for me on a HP ProBook 455 G7 with a Ryzen 5 4500U CPU and an integrated Radeon RX Vega 6 GPU. :( I hope, you are able to fix this, too. Keep up the great work, this tool has helped me a lot so far until I hit this AMD problem.

**Reply by PartialVolume** [reply]
Thanks for reporting that. How does it fail?

**Comment by Hellgore1974** [comment]
Hi, the screen goes black. Before it "hang" at booting ShredOS

Gesendet von Yahoo Mail auf Android 
 
  Am Do., Juni 22, 2023 at 12:35 schrieb ***@***.***>:   


Thanks for reporting that. How does it fail?

**Comment by PartialVolume** [comment]
Can you edit both grub.cfg files ( /boot/grub/grub.cfg and /EFI/BOOT/grub.cfg ) on the USB stick and add nomodeset as shown below. This will tell Linux to not switch between legacy frame buffer mode and the Linux Digital Rendering Manager (DRM).

If it then works we know it's a DRM issue with the graphics.

```
set default="0"
set timeout="0"

menuentry "shredos" {
	linux /boot/shredos console=tty3 loglevel=3 nomodeset
```

Thanks.

**Comment by Hellgore1974** [comment]
Hi, yeah I'll do that tomorrow. Thanx for the quick support 👍

Gesendet von Yahoo Mail auf Android 
 
  Am Do., Juni 22, 2023 at 16:48 schrieb ***@***.***>:   


Can you edit both grub.cfg files ( /boot/grub/grub.cfg and /EFI/BOOT/grub.cfg ) on the USB stick and add nomodeset as shown below. This will tell Linux to not switch between legacy frame buffer mode and the Linux Digital Rendering Manager (DRM).

If it then works we know it's a DRM issue with the graphics.
set default="0"
set timeout="0"

menuentry "shredos" {
	linux /boot/shredos console=tty3 loglevel=3 nomodeset

Thanks.

**Reply by PartialVolume** [reply]
>Hi, yeah I'll do that tomorrow. Thanx for the quick support +1

No problem

**Comment by Hellgore1974** [comment]
I can't seem to get a working IMG-File after umpacking, modifying the grub.cfg files and rebuilding it with imgburn. It always tells me: No bootfile found for UEFI. Maybe the image does not support x64 UEFI. I'm using Ventoy.

Gesendet von Yahoo Mail auf Android 
 
  Am Do., Juni 22, 2023 at 17:44 schrieb ***@***.***>:   



Hi, yeah I'll do that tomorrow. Thanx for the quick support +1


No problem

**Reply by PartialVolume** [reply]
I could be wrong, or misremembering, but I think there may have been some issue with imgburn. Rufus, I believe works fine. To be honest I'm 100% Linux so dd is the tool I would use. Maybe one of the guys who use windows can help out.

**Comment by Hellgore1974** [comment]
For the record: I also tried burning the image directly to an USB drive. This time the nomodeset change doesn't seem to have an impact: 

No further loading/booting

Gesendet von Yahoo Mail auf Android 
 
  Am Do., Juni 22, 2023 at 17:44 schrieb ***@***.***>:   



Hi, yeah I'll do that tomorrow. Thanx for the quick support +1


No problem

**Comment by Hellgore1974** [comment]
Hi there, sorry, couldn't get to it until now. I made the stick with Rufus and the Nomodeset added to both grub files. It boots to 100% but won't switch to the interface 🙁 Thanks again for the support, I really appreciate it!

Gesendet von Yahoo Mail auf Android 
 
  Am Fr., Juni 23, 2023 at 23:21 schrieb ***@***.***>:   


I could be wrong, or misremembering, but I think there may have been some issue with imgburn. Rufus, I believe works fine. To be honest I'm 100% Linux so dd is the tool I would use. Maybe one of the guys who use windows can help out.

**Reply by PartialVolume** [reply]
I need the dmesg.txt file from the USB stick to figure out what's going on with the graphics. Can you try the following:

1. **Important:** If you haven't already done so download this version of ShredOS [shredos-2021.08.2_23.3_x86-64_0.34_20230604.img](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2021.08.2_23.3_x86-64_0.34/shredos-2021.08.2_23.3_x86-64_0.34_20230604.img) this is the very latest pre-release. It's the only version that contains a feature that copies the dmesg output to the root directory of the USB stick as a file called dmesg.txt. So first thing is makesure you have this copy burned to your USB stick. Don't worry about editing the grub.cfg file, just leave them as default.
2. Boot the new ShredOS from your USB stick, don't remove the stick during this process as ShredOS will want to write to it.
3. When the boot process reaches 100%, **wait 30 seconds**. This 30 second wait allows the kernel to boot and nwipe to launch. You have to assume ShredOS is running even though you can't see anything happening on screen.
4. Important. Press CONTROL C which will abort Nwipe, (this assumes it's running even though you can't see what's going on)
5. **Wait 30 seconds** again. This is so ShredOS can mount the USB, write the dmesg.txt file and unmount the USB. The dmesg.txt only gets written to USB after nwipe exits, hence why we do a CONTROL C.
6. Remove the USB stick and check if it contains dmesg.txt. if it does post the contents here. That will tell us what's going on with the graphics.

**Reply by PartialVolume** [reply]
Just a quick note to avoid any confusion regarding pre-release versions. This announcement relates to the version **23.1** as in v2021.08.2_**23.1**_x86-64_0.34 which now isn't the latest pre-release. 

The latest pre-release as linked to in my previous comment and at the time of writing is **23.3** as in shredos-2021.08.2_**23.3**_x86-64_0.34_20230604.img. 23.3 which has the feature that auto writes the dmesg.txt file to the USB stick.

**Comment by PartialVolume** [comment]
Some system bioses support selection of multiple video card devices, often there is three settings, one or two devices and an auto mode. If you have this option in your bios try booting ShredOS with each setting and see if one particular one works. I had this issue with a Dell PC system that just got to 100% then blank screen as it had the incorrect selection in the bios. Not sure if that really applies to a laptop but worth looking in the bios anyway just to be sure.

**Comment by Hellgore1974** [comment]
Hi, I follow your instructions, but ShredOS won't create the dmesg.txt file. The newer version boots, screen turns black - I wait for a bit over half a minute then press Control+C and wait another minute. Then checking the UBS stick returns nothing. BIOS has no more GPU settings than allocating up to 512MB RAM to graphics. :(

**Comment by PartialVolume** [comment]
How easy would it be to install a distro like Ubuntu on this system?

The only thing I can think of is to install a Linux distro, to confirm it will run Linux then grab the dmesg from Ubuntu, assuming it installed and booted ok.

The Ubuntu dmesg would then give me an idea about what is being loaded, so I can check ShredOS isn't missing something.

**Comment by PartialVolume** [comment]
Very interesting thread about the same model here, well worth a read. https://forum.manjaro.org/t/impossible-installation-on-hp-455-g7-amd-renoir-4500u/26755/32

The final solution seems to be this https://forum.manjaro.org/t/impossible-installation-on-hp-455-g7-amd-renoir-4500u/26755/32

Can you make the same change in your bios and see if that works?

Quote
```
I hesitate to dare to say it’s done. I experienced too many funny things, but situation
is that I spent again some hours trying to figure out some useful Grub parameters.
To make the long story short: Useless.

Then I started to investigate the BIOS default settings one by one (this one has lot of options/not like Acer). After changing this from “all legacy” to “all UEFI” all started to work fine. Installation and booting system repeatedly.

Frankly speakin, I never looked at this “option rom” page. All I verified beforehand was Securuty (secure boot off), Boot, Options like that…
I first hesitated to set this as fixed, maybe I shoud have waited another day for that. BUT it looks great!!!
Now the work starts to put the data back tweak the system to the scheme I’m used to, as well as I hesitate here…
It was a great motivation [@Friedrich1](https://forum.manjaro.org/u/friedrich1) as w/o your statement that it shoud run, I would not spent so much effort to find out whats wrong. Thank you!
Thanks as well to [@megavolt](https://forum.manjaro.org/u/megavolt) and [@stephane](https://forum.manjaro.org/u/stephane) for ideas and guidance digging the problems
```

![image](https://github.com/PartialVolume/shredos.x86_64/assets/22084881/57dff56c-46bf-403a-b010-14db3e35603a)

**Comment by Hellgore1974** [comment]
:(
I tried all three settings for ROM Launch Policy - nothing helped. ShredOS boots to 100% and then Blackscreen. I even tried all possible combinations between legacy/UEFI boot and ROM policies. I really hoped, this would be the solution...

**Reply by PartialVolume** [reply]
Me too.

I was looking at ShredOS vs Ubuntu and in /usr/lib/firmware/amdgpu in Ubuntu there are 504 binaries while in ShredOS (/lib/firmware/amdgpu) there are 407. I can't find any other drivers except the fallback llvm. Without the dmesg I can't really figure out what is missing. Maybe when I update to the next buildroot, the buildroot guys will have included more amd drivers.

**Comment by Hellgore1974** [comment]
And no, I can't install Linux on this Laptop, sorry.

**Reply by PartialVolume** [reply]
You don't really need to install Ubuntu, just boot the live DVD then run dmesg > dmesg.txt.

**Comment by PartialVolume** [comment]
@Hellgore1974 Ref. AMDGPU [firmware](https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/) 

With the latest version of ShredOS I'm pulling in **LINUX_FIRMWARE_VERSION = 20210511**

I can hopefully update this to **LINUX_FIRMWARE_VERSION = [20230625](https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tag/?h=20230625)** without doing a full buildroot upgrade. This version contains the missing 100+ missing AMDGPU binaries.

I'll let you know as soon as I've created a new pre-release.

**Comment by Hellgore1974** [comment]
Wow, that sounds really cool! Thank you again for all your work and support. ShredOS is simply the best secure erasing app out there and it would be great if I can continue to use it 👍

Gesendet von Yahoo Mail auf Android 
 
  Am Sa., Juli 1, 2023 at 16:03 schrieb ***@***.***>:   


@Hellgore1974 Ref. AMDGPU firmware

With the latest version of ShredOS I'm pulling in LINUX_FIRMWARE_VERSION = 20210511

I can hopefully update this to LINUX_FIRMWARE_VERSION = 20230625 without doing a full buildroot upgrade. This version contains the missing 100+ missing AMDGPU binaries.

I'll let you know as soon as I've created a new pre-release.

**Comment by PartialVolume** [comment]
@Hellgore1974 There you go, a pre-release with the very latest firmware [v2021.08.2_23.4_x86-64_0.34]( https://github.com/PartialVolume/shredos.x86_64/releases/tag/v2021.08.2_23.4_x86-64_0.34)

If you could give that a try, I've got my fingers crossed.

I've tested on the following laptops with AMD & Nvidia graphics and they all boot ok:

- AMD Radeon HD8330 (Kabini) (Packard Bell Easynote TE69KB)
- AMD Radeon HD6320 (Wrestler) (HP Pavilion DM1)
- AMD Radeon HD3450/3470 (Sony Vaio)

and also on Nvidia hardware.

- Nvidia GT218 (NVS 300) (Dell optiplex 9010)
- Nvidia GeForce RTX 3060 (Clevo based Laptop)

**Comment by Hellgore1974** [comment]
Hi, I'm afraid I can't report a success. Still black screen after booting to 100%. And I tried both new versions 23.4 and 23.5; set both legacy and uefi-boot

**Comment by PartialVolume** [comment]
@Hellgore1974 the 23.5 was was aimed more at the Dell servers with the Matrix G200 hardware, I wasn't expecting much for your HP ProBook 455 G7 with a Ryzen 5 4500U CPU.

I was hoping 23.4 would fix your HP. What OS does it normally boot?

**Comment by Hellgore1974** [comment]
Hi, was just a desperate try with 23.5 :) 
The system is normally a Windows 10 machine. It seems that the Radeon RX Vega 6 firmware isn't included if your tests with Radeon HD gpus were successful.

**Comment by PartialVolume** [comment]
@Hellgore1974 What I find interesting about the HP ProBook 455 G7 with a Ryzen 5 4500 is just how many others have experienced the black screen just after boot. Doesn't seem to matter what Linux distro you use. Sometimes they will get it working with setting UEFI boot only in the bios, sometimes they put nomodeset on the Linux boot line in grub.cfg. Others might switch distro and find it works, even saw a Ubuntu user upgrade Ubuntu and get the problem which made me think there was a regression in the code.

It's a very odd thing. Can the you try the nomodeset command with 23.5. I would append nomodeset to the end of the Linux boot line in grub.cfg. in both grub files /boot/grub/ and /EFI/BOOT/. This will disable the switch from simple frame buffer support to the Linux DRM manager.

One other thought. It just occured to me another possibility. Maybe due to an ACPI issue the laptops LED backlight is being switched off when Linux switches to DRM. If the laptop has a VGA port any possibility of hooking up a monitor to it to see if that works ok. When the screen goes black does it really go completely black as if the laptop was switched off or can you see a slight glow from the screen. You might need a dark room for that.

**Comment by Hellgore1974** [comment]
Feedbackreport: 
- When I add nomodeset to the grubs the boot screen hangs at 100% and there's no reaction whatsoevert to pressed key (ctrl+c, e.g.)
- Without nomodeset the boot switches to the black screen but the power light is on, the keyboard backlight fades out but reacts to pressed keys, turning back on and the monitor panel backlight is active, so it is a black screen but not turned off.

**Reply by PartialVolume** [reply]
At least we know it's not the backlight being switched off.

Can you check the bios revision? There is a lot of fixes including new video firmware in the latest version. Just wondering if that's up to date.

https://support.hp.com/gb-en/drivers/selfservice/swdetails/hp-probook-455-g7-notebook-pc/32780046/swItemId/ob-310470-1

![Screenshot_20230703_103742](https://github.com/PartialVolume/shredos.x86_64/assets/22084881/9578e7ec-87ba-4d96-88fe-7b0a51b7c291)

**Comment by PartialVolume** [comment]
Is it correct that the HP Probook 455 has docking station capability? I was just wondering if the display is coming up on the wrong port. I've seen that happen before on Linux. If there is a VGA port or maybe display port output then a second monitor connected to it would either rule that out or confirm it.

**Comment by Hellgore1974** [comment]
Hi, further feedbacks:
- Upgraded Firmware from 1.12 to 1.13 but same behaviour either with or without nomodeset
- I can try with a dockingstation as I happen to have one with VGA, the laptop itself has only hdmi as external port. I'll report back :)

**Comment by Hellgore1974** [comment]
Sorry to be the bringer of bad news, but neither VGA nor DisplayPort via Dockingstation brought a different result both with and without nomodeset-grubmodification.

**Comment by Hellgore1974** [comment]
Would this be of any help?
There are 2 Radeon RX Vega 6 Mobile4 Ubuntu Bionic Driver available Alternatives: the AMDGPU Proprietary and the Open-Source.

**Comment by PartialVolume** [comment]
ShredOS doesn't make use of the x-org drivers, it first uses simple framebuffer support then the kernel switches to direct renderering manager (DRM) that handles the GPU. AMDGPU provides the DRM driver for AMD GPU's. That's as far as ShredOS goes, it doesn't need xorg drivers for a full window desktop experience. I hope that I don't need to install extra software just to get the AMD DRM to work properly as that starts to really add to the size of the build. It sort of wouldn't make sense to have to install that extra stuff just to get DRM for AMD working.

What puzzles me is the nodemodeset command not seeming to work. This disables the kernel switching to DRM so Nwipe stays working with the simple framebuffer. However having said that I'm assuming you never see any text flash up on the screen even for a second which would suggest it never even manages to get the simple framebuffer working. That would imply it's failing very early in the boot. Do you see any text flash onto the screen momentarily before going blank?

Here's all the DRM software, including AMDGPU enabled in ShredOS
 
![Screenshot_20230703_125150](https://github.com/PartialVolume/shredos.x86_64/assets/22084881/a851edf4-9aee-4d89-b066-fb3756ecf827)

I think what maybe useful, is to boot this system of a Ubuntu live USB stick or DVD. You don't need to install anything, just boot it to the desktop, then open a terminal and run dmesg. i.e `dmesg > dmesg.txt` then copy the file to USB and post it here. Assuming Ubuntu boots in the first place this would tell me what drivers Ubuntu is using, especially in the early stages of the boot where it goes from framebuffer to DRM before switching to x-org.

**Comment by PartialVolume** [comment]
One other thing, can you try this version '21' [shredos-2021.08.2_21_x86-64_0.32.023_20220123.img](https://github.com/PartialVolume/shredos.x86_64/releases/download/v2021.08.2_21_x86-64_0.32.023/shredos-2021.08.2_21_x86-64_0.32.023_20220123.img)

This version is an older version from Jan 2022. This version doesn't use DRM software and keeps ShredOS & nwipe operating in the simple frame buffer mode. I just want to see if that works. If it does work, then it would indicate a DRM issue specific to that GPU that doesn't affect other AMD GPUs.

**Comment by Hellgore1974** [comment]
- The alternative version also hangs after booting to 100%, not even switching to black screen.
- Ubuntu boots with no problem to desktop - here's the dmesg-file
[dmesg.txt](https://github.com/PartialVolume/shredos.x86_64/files/11936882/dmesg.txt)

**Reply by PartialVolume** [reply]
Thanks.

Just putting the graphics related sections of dmesg here for later reference:

The kernel seems to be loading some firmware into presumably the GPU, VCN & DMUB using PSP.. what ever that means. However that 4 seconds into the boot, the simple/VGA/EFI frame buffer text should have already been displayed before it switches to DRM.

EFI frame buffer is working because we see the Grub boot text and percentage, it should then switch to simple/VGA frame buffer which doesn't happen. (It does on Ubuntu)

I did wonder if I was missing vgaarb, however booting ShredOS on another AMD system shows vgaarb gets loaded.

I'll have to study this a bit more and get back to you.

Having booted of Ubuntu you could still run nwipe from there of course with a `sudo apt install nwipe; sudo nwipe` I think Ubuntu are shipping 0.34 since 22.10.

```
[    0.328492] pci 0000:05:00.0: vgaarb: setting as boot VGA device
[    0.328492] pci 0000:05:00.0: vgaarb: bridge control possible
[    0.328492] pci 0000:05:00.0: vgaarb: VGA device added: decodes=io+mem,owns=none,locks=none
[    0.328492] vgaarb: loaded
```

```
[    4.369771] [drm] Loading DMUB firmware via PSP: version=0x0101001F
[    4.370433] [drm] Found VCN firmware Version ENC: 1.16 DEC: 5 VEP: 0 Revision: 3
[    4.370437] amdgpu 0000:05:00.0: amdgpu: Will use PSP to load VCN firmware
```

```
[    0.102950] Console: colour dummy device 80x25
[    0.102968] printk: console [tty0] enabled

[    0.328492] pci 0000:05:00.0: vgaarb: setting as boot VGA device
[    0.328492] pci 0000:05:00.0: vgaarb: bridge control possible
[    0.328492] pci 0000:05:00.0: vgaarb: VGA device added: decodes=io+mem,owns=none,locks=none
[    0.328492] vgaarb: loaded

[    0.770628] efifb: probing for efifb
[    0.770646] efifb: showing boot graphics
[    0.771801] efifb: framebuffer at 0xd0000000, using 8100k, total 8100k
[    0.771802] efifb: mode is 1920x1080x32, linelength=7680, pages=1
[    0.771803] efifb: scrolling: redraw
[    0.771804] efifb: Truecolor: size=8:8:8:8, shift=24:16:8:0
[    0.771875] Console: switching to colour frame buffer device 240x67
[    0.773312] fb0: EFI VGA frame buffer device
[    0.773315] EFI Variables Facility v0.08 2004-May-17

[    4.364798] [drm] amdgpu kernel modesetting enabled.
[    4.367160] amdgpu: Virtual CRAT table created for CPU
[    4.367170] amdgpu: Topology: Add CPU node
[    4.367328] [drm] initializing kernel modesetting (RENOIR 0x1002:0x1636 0x103C:0x8730 0xC3).
[    4.367462] [drm] register mmio base: 0xE0600000
[    4.367463] [drm] register mmio size: 524288
[    4.368839] [drm] add ip block number 0 <soc15_common>
[    4.368840] [drm] add ip block number 1 <gmc_v9_0>
[    4.368842] [drm] add ip block number 2 <vega10_ih>
[    4.368843] [drm] add ip block number 3 <psp>
[    4.368843] [drm] add ip block number 4 <smu>
[    4.368844] [drm] add ip block number 5 <dm>
[    4.368845] [drm] add ip block number 6 <gfx_v9_0>
[    4.368846] [drm] add ip block number 7 <sdma_v4_0>
[    4.368847] [drm] add ip block number 8 <vcn_v2_0>
[    4.368848] [drm] add ip block number 9 <jpeg_v2_0>
[    4.368864] amdgpu 0000:05:00.0: amdgpu: Fetched VBIOS from VFCT
[    4.368866] amdgpu: ATOM BIOS: 113-RENOIR-037
[    4.368899] [drm] VCN decode is enabled in VM mode
[    4.368900] [drm] VCN encode is enabled in VM mode
[    4.368900] [drm] JPEG decode is enabled in VM mode
[    4.368902] checking generic (d0000000 7e9000) vs hw (d0000000 10000000)
[    4.368903] fb0: switching to (null) from EFI VGA
[    4.369043] Console: switching to colour dummy device 80x25
[    4.369110] amdgpu 0000:05:00.0: vgaarb: deactivate vga console
[    4.369112] amdgpu 0000:05:00.0: amdgpu: Trusted Memory Zone (TMZ) feature enabled
[    4.369113] amdgpu 0000:05:00.0: amdgpu: PCIE atomic ops is not supported
[    4.369119] amdgpu 0000:05:00.0: amdgpu: MODE2 reset
[    4.369242] [drm] vm size is 262144 GB, 4 levels, block size is 9-bit, fragment size is 9-bit
[    4.369247] amdgpu 0000:05:00.0: amdgpu: VRAM: 512M 0x000000F400000000 - 0x000000F41FFFFFFF (512M used)
[    4.369249] amdgpu 0000:05:00.0: amdgpu: GART: 1024M 0x0000000000000000 - 0x000000003FFFFFFF
[    4.369250] amdgpu 0000:05:00.0: amdgpu: AGP: 267419648M 0x000000F800000000 - 0x0000FFFFFFFFFFFF
[    4.369255] [drm] Detected VRAM RAM=512M, BAR=512M
[    4.369256] [drm] RAM width 128bits DDR4
[    4.369291] [drm] amdgpu: 512M of VRAM memory ready
[    4.369292] [drm] amdgpu: 7697M of GTT memory ready.
[    4.369299] [drm] GART: num cpu pages 262144, num gpu pages 262144
[    4.369420] [drm] PCIE GART of 1024M enabled.
[    4.369421] [drm] PTB located at 0x000000F400900000
[    4.369591] amdgpu 0000:05:00.0: amdgpu: PSP runtime database doesn't exist
[    4.369593] amdgpu 0000:05:00.0: amdgpu: PSP runtime database doesn't exist
[    4.369771] [drm] Loading DMUB firmware via PSP: version=0x0101001F
[    4.370433] [drm] Found VCN firmware Version ENC: 1.16 DEC: 5 VEP: 0 Revision: 3
[    4.370437] amdgpu 0000:05:00.0: amdgpu: Will use PSP to load VCN firmware

[    5.071917] [drm] reserve 0x400000 from 0xf41fb00000 for PSP TMR

[    5.158386] amdgpu 0000:05:00.0: amdgpu: RAS: optional ras ta ucode is not available
[    5.169845] amdgpu 0000:05:00.0: amdgpu: RAP: optional rap ta ucode is not available
[    5.169846] amdgpu 0000:05:00.0: amdgpu: SECUREDISPLAY: securedisplay ta ucode is not available
[    5.171060] amdgpu 0000:05:00.0: amdgpu: SMU is initialized successfully!
[    5.171999] [drm] Display Core initialized with v3.2.187!
[    5.172549] [drm] DMUB hardware initialized: version=0x0101001F
[    5.389884] [drm] DP Alt mode state on HPD: 1
[    5.511480] [drm] DM_MST: Differing MST start on aconnector: 00000000d5550bb5 [id: 89]
[    5.513442] [drm] kiq ring mec 2 pipe 1 q 0
[    5.517441] [drm] VCN decode and encode initialized successfully(under DPG Mode).
[    5.517463] [drm] JPEG decode initialized successfully.
[    5.519910] kfd kfd: amdgpu: Allocated 3969056 bytes on gart
[    5.520049] amdgpu: sdma_bitmap: 3
[    5.537392] memmap_init_zone_device initialised 131072 pages in 0ms
[    5.537401] amdgpu: HMM registered 512MB device memory
[    5.537439] amdgpu: SRAT table not found
[    5.537439] amdgpu: Virtual CRAT table created for GPU
[    5.537532] amdgpu: Topology: Add dGPU node [0x1636:0x1002]
[    5.537536] kfd kfd: amdgpu: added device 1002:1636
[    5.537609] amdgpu 0000:05:00.0: amdgpu: SE 1, SH per SE 1, CU per SH 8, active_cu_number 6
[    5.537681] amdgpu 0000:05:00.0: amdgpu: ring gfx uses VM inv eng 0 on hub 0
[    5.537682] amdgpu 0000:05:00.0: amdgpu: ring comp_1.0.0 uses VM inv eng 1 on hub 0
[    5.537683] amdgpu 0000:05:00.0: amdgpu: ring comp_1.1.0 uses VM inv eng 4 on hub 0
[    5.537684] amdgpu 0000:05:00.0: amdgpu: ring comp_1.2.0 uses VM inv eng 5 on hub 0
[    5.537685] amdgpu 0000:05:00.0: amdgpu: ring comp_1.3.0 uses VM inv eng 6 on hub 0
[    5.537686] amdgpu 0000:05:00.0: amdgpu: ring comp_1.0.1 uses VM inv eng 7 on hub 0
[    5.537687] amdgpu 0000:05:00.0: amdgpu: ring comp_1.1.1 uses VM inv eng 8 on hub 0
[    5.537688] amdgpu 0000:05:00.0: amdgpu: ring comp_1.2.1 uses VM inv eng 9 on hub 0
[    5.537689] amdgpu 0000:05:00.0: amdgpu: ring comp_1.3.1 uses VM inv eng 10 on hub 0
[    5.537690] amdgpu 0000:05:00.0: amdgpu: ring kiq_2.1.0 uses VM inv eng 11 on hub 0
[    5.537691] amdgpu 0000:05:00.0: amdgpu: ring sdma0 uses VM inv eng 0 on hub 1
[    5.537691] amdgpu 0000:05:00.0: amdgpu: ring vcn_dec uses VM inv eng 1 on hub 1
[    5.537693] amdgpu 0000:05:00.0: amdgpu: ring vcn_enc0 uses VM inv eng 4 on hub 1
[    5.537693] amdgpu 0000:05:00.0: amdgpu: ring vcn_enc1 uses VM inv eng 5 on hub 1
[    5.537694] amdgpu 0000:05:00.0: amdgpu: ring jpeg_dec uses VM inv eng 6 on hub 1
[    5.553971] [drm] Initialized amdgpu 3.47.0 20150101 for 0000:05:00.0 on minor 0
[    5.559404] fbcon: amdgpudrmfb (fb0) is primary device
[    5.559537] [drm] DSC precompute is not needed.

[    5.603289] Console: switching to colour frame buffer device 240x67
[    5.621598] amdgpu 0000:05:00.0: [drm] fb0: amdgpudrmfb frame buffer device

[    5.942379] [drm] Downstream port present 1, type 0

[    6.064256] [drm] Downstream port present 1, type 0

```

**Comment by PartialVolume** [comment]
Can you try one more thing, append `acpi=off` to the end of the linux boot command in both grub.cfg in ShredOS. Try booting and see if any text appears this time. If not remove `acpi=off` from the grub files. I just want to see if acpi is somehow killing the kernel early on. On my AMD with acpi=off nwipe can't detect drives so it's not something you want to leave in grub, however for the purposes of this test it might tell us it's a acpi issue if we then get some text on screen.

Also, I don't remember if I've already asked you this, but is there any graphics related settings in the bios, they might be under acpi or maybe somewhere else.

**Comment by PartialVolume** [comment]
If that doesn't get us anywhere, the only thing I can think of is to wait until I do a full upgrade of ShredOS to the latest kernel & drivers.

**Comment by PartialVolume** [comment]
One final thing, are you in a position that you can carry out [these](https://github.com/PartialVolume/shredos.x86_64#how-to-wipe-drives-on-headless-systems-or-systems-with-faulty-display-hardware-for-use-on-secure-lans-only) instructions.

In the off chance the kernel is getting far enough to enable telnetd then there is the possibility to log in to ShredOS via the network and run dmesg.

**Reply by PartialVolume** [reply]
The laptop would need to be connected to a network with a RJ45/Cat5 network connection  (WiFi doesn't work) and the network needs a router with DHCP enabled as ShredOS will request a IP address.

**Comment by PartialVolume** [comment]
Another black screen problem I found with the 5.13 kernel as used in ShredOS, `black screen on AMD GPU.` https://bbs.archlinux.org/viewtopic.php?id=267874 

The workaround was adding `amdgpu.dc=0` to grub. May be worth trying that. The ultimate fix was to upgrade the kernel.

**Comment by Hellgore1974** [comment]
Hi, I'll try that tomorrow and report back. Thx

Gesendet von Yahoo Mail auf Android 
 
  Am Mo., Juli 3, 2023 at 17:37 schrieb ***@***.***>:   


Another black screen problem I found with the 5.13 kernel as used in ShredOS, black screen on AMD GPU. https://bbs.archlinux.org/viewtopic.php?id=267874

The workaround was adding amdgpu.dc=0 to grub. May be worth trying that. The ultimate fix was to upgrade the kernel.

**Comment by PartialVolume** [comment]
I'm building ShredOS on the very latest buildroot 2023.05 and using the very latest kernel, v6.3.

I'm not sure how long it will take, could be a day, could be a week. All depends on how many problems I have to solve.

It's about time I upgraded ShredOS as the current kernel is nearly two years old, so will be interesting to see if it solves the blank screen problem on your HP ProBook 455 G7 / Ryzen 5 4500 GPU

**Comment by Hellgore1974** [comment]
Just reporting back the latest tryouts:
- amdgpu.dc=0 just produces some colourful lines and pixels but nothing more
- acpi=off doesn't seem to change anything except for letting the screen "hang" at boot after reaching 100%, just like with nomodeset added to grub.
- I can not try out the network test

I'll be happily waiting for your kernelupgrade to ShredOS.

**Comment by PartialVolume** [comment]
Thanks for letting me know. I'm currently building the new ShredOS 2023.05. This version will build both legacy and EFI, previous versions used a ShredOS script the produce the same effect so I need to make sure that works correctly. The build I did yesterday was way too large at 800MB, without going into all the details the buildroot conversion process added stuff that is really not needed, like gstreamer support ?? So once I've gone through all the packages and got the size back down to less than 180MB and checked the packages I do need are still there then I'll let you know.

**Comment by PartialVolume** [comment]
@Hellgore1974 See #155

**Comment by gkharrison** [comment]
Some fantastic changes coming in next version. Hopefully it will be released soonish?

**Comment by PartialVolume** [comment]
@gkharrison very soon, yes. Just finishing of the gui set system time feature. As usual and unfortunately other things in life are taking priority, but as soon I've finished the time feature and that unusual segfault reported yesterday in nwipe, I'll do a release.

**Comment by MallocArray** [comment]
Tried the latest public release on an old Toshiba Satellite C55D laptop (AMD A8 processor and maybe video card?) using the USB image in both CSM and UEFI modes and after saying it was loading shredos, it would just hang and never finish, but the white text was still on the black screen.

Tried the latest pre-release 23.5 and it worked just fine in UEFI mode.

**Reply by PartialVolume** [reply]
Thanks for letting me know. Much appreciated.
